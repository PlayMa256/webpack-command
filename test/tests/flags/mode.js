const { apply, build, crcDist, test, validate } = require('../../util');

test('--mode', module, () => {
  for (const name of ['dev', 'prod']) {
    const fixture = `mode/${name}`;
    const opts = { fixture };

    let config;

    it(`${name} should validate`, () => {
      expect(validate(opts)).toEqual(true);
    });

    it(`${name} should apply`, () => {
      config = apply(opts);

      expect(config).toMatchSnapshot();
    });

    it(`${name} should build`, () =>
      build(config).then((result) => {
        expect(result).toMatchSnapshot();
        expect(crcDist()).toMatchSnapshot();
      }));
  }
});
