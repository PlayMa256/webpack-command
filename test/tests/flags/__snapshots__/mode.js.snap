// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flags > --mode > dev should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --mode > dev should build #0 1`] = `
"Version: webpack 4.6.0
  Asset      Size  Chunks             Chunk Names
main.js  2.96 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 41 bytes {main} [built]"
`;

exports[`Flags > --mode > dev should build #1 1`] = `"bf58edd8"`;

exports[`Flags > --mode > prod should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "production",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --mode > prod should build #0 1`] = `
"Version: webpack 4.6.0
  Asset       Size  Chunks             Chunk Names
main.js  641 bytes       0  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 41 bytes {0} [built]"
`;

exports[`Flags > --mode > prod should build #1 1`] = `"969d6c1f"`;
